prospero-common/src/main/java/org/wildfly/prospero/api/ArtifactChange.java

```diff
@@ -22,44 +22,25 @@
 
 import java.util.Optional;
 
-public class ArtifactChange {
-    private Artifact oldVersion;
-    private Artifact newVersion;
+public class ArtifactChange extends Diff {
 
     public ArtifactChange(Artifact oldVersion, Artifact newVersion) {
-        this.oldVersion = oldVersion;
-        this.newVersion = newVersion;
+        super(toGav(oldVersion == null?newVersion:oldVersion), oldVersion==null?null:oldVersion.getVersion(), newVersion==null?null:newVersion.getVersion());
     }
 
     public String getArtifactName() {
-        if (oldVersion == null) {
-            return toGav(newVersion);
-        } else {
-            return toGav(oldVersion);
-        }
+        return getName().get();
     }
 
     public Optional<String> getOldVersion() {
-        return oldVersion==null?Optional.empty():Optional.of(oldVersion.getVersion());
+        return getOldValue();
     }
 
     public Optional<String> getNewVersion() {
-        return newVersion==null?Optional.empty():Optional.of(newVersion.getVersion());
-    }
-
-    @Override
-    public String toString() {
-        if (oldVersion == null) {
-            return String.format("Install [%s]:\t\t [] ==> %s", toGav(newVersion), newVersion.getVersion());
-        }
-        if (newVersion == null) {
-            return String.format("Remove [%s]:\t\t %s ==> []", toGav(oldVersion), oldVersion.getVersion());
-        }
-        final String gac = toGav(oldVersion);
-        return String.format("Update [%s]:\t\t %s ==> %s", gac, oldVersion.getVersion(), newVersion.getVersion());
+        return getNewValue();
     }
 
-    private String toGav(Artifact artifact) {
+    private static String toGav(Artifact artifact) {
         final String gac;
         if (artifact.getClassifier() == null || artifact.getClassifier().isEmpty()) {
```

Comment:
I'm wondering why the classifier is part of this string? (Rather then e.g. version.)